test/tools/libtesteth/TestOutputHelper.cpp

```diff
@@ -129,30 +129,40 @@ void TestOutputHelper::checkUnfinishedTestFolders()
     // -t SuiteName/caseName   parse caseName as filter
     // rCurrentTestSuite is empty if run without -t argument
     string filter;
-    size_t pos = Options::get().rCurrentTestSuite.find('/');
+    size_t pos = Options::get().rCurrentTestSuite.rfind('/');
     if (pos != string::npos)
         filter = Options::get().rCurrentTestSuite.substr(pos + 1);
 
+    bool notAFilter = false;
     if (!filter.empty())
     {
-        if (m_finishedTestFoldersMap.size() > 1)
-        {
-            std::cerr << "ERROR: Expected a single test to be passed: " << filter << "\n";
-            return;
-        }
-
         // Unit tests does not mark test folders
         if (m_finishedTestFoldersMap.size() == 0)
             return;
 
         std::map<boost::filesystem::path, FolderNameSet>::const_iterator it =
             m_finishedTestFoldersMap.begin();
-        if (!pathHasTests(it->first / filter))
-            std::cerr << "WARNING: Test folder " << it->first / filter
-                      << " appears to have no tests!"
-                      << "\n";
+        if (!boost::filesystem::exists(it->first / filter))
+            notAFilter = true;
```

Comment:
Is it the case when `-t` points to a unit test?

Reply:
Ah no, for unit tests it should return from line 141.
When this is the case then?

Reply:
it is the case when you select  `-t TestSuite / SubTestSuite`

Reply:
It looks like you could move this check to the outer `if` (line 137)
Kind of like
```
if (!filter.empty() || boost::filesystem::exists(m_finishedTestFoldersMap.begin()->first / filter))
{
...
}
else
{
...
}
```
(but check that `m_finishedTestFoldersMap` is not empty first)

Would it be simpler?

Reply:
Also maybe it would easier/more efficient to look at `c_allTestNames` to check whether it's single folder or subsuite, but I'm not sure

Reply:
if you want to call c_allTestNames more then from one place, will have to intoduce c_allTestNames.cpp